src/main/java/org/spongepowered/api/event/conversation/ConversationOpenEvent.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.event.conversation;
+
+import org.spongepowered.api.command.conversation.Conversant;
+import org.spongepowered.api.command.conversation.Conversation;
+import org.spongepowered.api.command.conversation.ConversationArchetype;
+import org.spongepowered.api.event.Cancellable;
+
+import java.util.Set;
+
+/**
+ * An event called when a {@link Conversation} is either starting or has started.
+ *
+ * <p>You really should use the sub-interfaces as they include the most access
+ * to information and let you know if the conversation is starting or has
+ * started already.</p>
+ */
+public interface ConversationOpenEvent extends ConversationEvent {
+
+    /**
+     * Called before the {@link Conversation} is created, giving you access to
+     * the archetype as well as who will be in the conversation.
+     *
+     * <p>If you are removing all conversants, it is better to simply cancel the
+     * conversation creation.</p>
```

Comment:
Rather than this and leave a zero conversant conversation, why not state that you'll detect this and just not create the conversation if it is empty?